src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java

```diff
@@ -46,10 +51,23 @@ public final void setConnectionValues(BitBucket api, ConnectionValues values) {
 
     @Override
     public final UserProfile fetchUserProfile(BitBucket api) {
-        BitBucketUser user = api.userOperations().getUserWithRepositories()
-                .getUser();
+        BitBucketUser user = api.userOperations().getUserWithRepositories().getUser();
+        String primaryEmail = null;
+        try {
+            List<BitBucketEmailAddress> emailAddresses = api.usersOperations()
+                    .emailsOperations().getListOfUserEmailAddresses(user.getUsername());
+            for (BitBucketEmailAddress emailAddress : emailAddresses) {
+                if (emailAddress.getPrimary()) {
+                    primaryEmail = emailAddress.getEmail();
+                }
+            }
+        } catch (HttpClientErrorException clientError) {
+            if (!HttpStatus.FORBIDDEN.equals(clientError.getStatusCode())) {
```

Comment:
if Permissions:Acount:Email is not set, then email property is not expected in UserProfile by client (bitbucket will answer with 403). Every other error will be pushed up. Warning implemented.